### PR TITLE
matches sdk to os or fails. fix #1319

### LINKED ISSUE #1319: on mac use sdk that match os
if user has sdks that are different from their os cmake gets confused onto which is best to use.

### PR DESCRIPTION
**Pull request by doutriaux1**
@aashish24 @remram44 please review. @remram44 please put all of your sdks back in as before and let me know if it still fails building.

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 looks like you might be missing some baselines https://open.cdash.org/testDetails.php?test=335626893&build=3810495

**Comment by doutriaux1** [comment]
thanks @durack1 https://github.com/UV-CDAT/uvcdat-testdata/pull/49 hadnt been merged in after merging the PR. @aashish24 please be careful about this when merging. Thanks.

**Comment by durack1** [comment]
@doutriaux1 want to tweak the PR to initialize a rebuild?

**Comment by doutriaux1** [comment]
nah...

File: CMakeLists.txt [hunk 1]
```diff
@@ -119,26 +119,25 @@ if(APPLE)
...
-      set(CMAKE_C_COMPILER clang)
-      set(CMAKE_CXX_COMPILER clang++)
-      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
-      set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10 CACHE STRING "" FORCE)
-      set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk" CACHE PATH "" FORCE)
+  if ((${OSX_VER_MAJOR} VERSION_EQUAL 8) AND (HAVE_10_8_SDK))
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.8 CACHE STRING "" FORCE)
+    set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk" CACHE PATH "" FORCE)
+  elseif((${OS_X_VER_MAJOR} VERSION_EQUAL 9) AND (HAVE_10_9_SDK))
```
Comment (aashish24) [inline CMakeLists.txt]: I think its perfectly valid to use 10.8 SDK on say Mac 10.9. But if this works better then I am fine with that.

**Comment by aashish24** [comment]
LGTM

**Pull request merged**